### PR TITLE
evmrs: go integration

### PR DESCRIPTION
**Pull request by LorenzSchueler**
- build evmrs using make by invoking `cargo`
- integrate emvrs into Tosca
- include .cargo/bin in PATH for Jenkins

### CODE REVIEW

**Review by simonlechner — CHANGES_REQUESTED** [review]
Looks good.
Please update requirements and build instructions since you are changing the Makefile.

File: go/interpreter/evmrs/ct_test.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	"github.com/Fantom-foundation/Tosca/go/ct/st"
+	"github.com/Fantom-foundation/Tosca/go/tosca"
+	"github.com/Fantom-foundation/Tosca/go/tosca/vm"
+)
+
+func TestCtAdapter_Add(t *testing.T) {
+	s := st.NewState(st.NewCode([]byte{
+		byte(vm.PUSH1), 3,
+		byte(vm.PUSH1), 4,
+		byte(vm.ADD),
```
[the remark this answers is not in the document]
Comment (LorenzSchueler) [reply go/interpreter/evmrs/ct_test.go:27]: Sure. However, I just copied it from evmzero. Should I leave it as is for consistency or update both?

File: Makefile [hunk 2]
```diff
@@ -11,14 +11,14 @@ TOSCA_CPP_ASSERT = ON
...
-        bench bench-go clean clean-go clean-cpp evmone evmone-clean license-headers
+.PHONY: all tosca tosca-go tosca-cpp tosca-rust test test-go test-cpp test-rust test-cpp-asan \
+        bench bench-go clean clean-go clean-cpp clean-rust evmone evmone-clean license-headers
 
 all: tosca
 
 tosca: tosca-go
 
-tosca-go: tosca-cpp evmone
+tosca-go: tosca-cpp tosca-rust evmone
```
[the remark this answers is not in the document]
Comment (LorenzSchueler) [reply Makefile:21]: The instructions are already in `BUILD.md`, but they were outdated. Are updated now.

File: go/interpreter/evmrs/evmrs_test.go
[the remark this answers is not in the document]
Comment (LorenzSchueler) [reply go/interpreter/evmrs/evmrs_test.go:1]: They are just copied from `evmzero`.

File: go/interpreter/evmrs/ct_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (LuisPH3) [reply go/interpreter/evmrs/ct_test.go:27]: Don't update evmzero, you can define your own tests in this namespace. Evmzero is pending a large refactor and testing effort, some day.

**Review by LuisPH3 — COMMENTED** [review]
I got cold feet with this, let me check first what are the implications. Since aida calls our makefile, it will require rust everywhere.

File: go/interpreter/evmrs/ct_test.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+		t.Fatalf("unexpected status: wanted %v, got %v", want, got)
+	}
+
+	if want, got := cc.NewU256(3+4), s.Stack.Get(0); !want.Eq(got) {
+		t.Errorf("unexpected result: wanted %s, got %s", want, got)
+	}
+}
+
+func TestCtAdapter_Interface(t *testing.T) {
+	// Compile time check that ctAdapter implements the st.Evm interface.
```
Comment (HerbertJordan) [inline go/interpreter/evmrs/ct_test.go:53]:
```suggestion
	// Compile time check that ctAdapter implements the ct.Evm interface.
```

File: go/interpreter/evmrs/evmrs_test.go [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+		t.Fatalf("running the fib example failed: %v", err)
+	}
+
+	if got.Result != wanted {
+		t.Fatalf("unexpected result, wanted %v, got %v", wanted, got.Result)
+	}
+}
+
+func TestEvmrs_DumpProfile(t *testing.T) {
+	t.Skip("evmrs does not implement the profiling configuration yet") // TODO
```
[the remark this answers is not in the document]
Comment (HerbertJordan) [reply go/interpreter/evmrs/evmrs_test.go:46]: Right, if it is not supported yet, just drop the test.

If you want to make sure it is not supported, have a test like
```
        instance, err := tosca.NewInterpreter("evmrs")
	if err != nil {
		t.Fatalf("failed to load evmrs interpreter: %v", err)
	}
	_, ok := instance.(tosca.ProfilingInterpreter)
	if ok {
		t.Fatalf("evmrs should not support profiling")
	}
```

**Review by HerbertJordan — APPROVED** [review]
The Go integration code looks good to me. I also managed to build and run it on my local system. So from my side it is an approval.

However, please work with @LuisPH3 and @simonlechner on making sure build dependencies are covered on Jenkins.

**Review by simonlechner — DISMISSED** [review]
As soon as we know that this wont break anybodies build it good to merge from my side.

**Comment by LuisPH3** [comment]
I have been checking what can fail once this pr is merged,
Pipelines in https://github.com/Fantom-foundation/LaScala/tree/main/tosca will stop working because two reasons: 

- [ ] GCP image does not have rust:  both "quick" and "db-small-ssd" have rust installed 
- [ ] pipeline does not add rust to the jenkins user path.

**Pull request merged**